java/runtime/src/main/java/org/ray/runtime/task/ArgumentsBuilder.java

```diff
@@ -32,27 +28,17 @@
   /**
    * Convert real function arguments to task spec arguments.
    */
-  public static List<FunctionArg> wrap(Object[] args, Language language, boolean isDirectCall) {
+  public static List<FunctionArg> wrap(Object[] args, Language language) {
     List<FunctionArg> ret = new ArrayList<>();
     for (Object arg : args) {
       ObjectId id = null;
       NativeRayObject value = null;
       if (arg instanceof RayObject) {
-        if (isDirectCall) {
-          throw new IllegalArgumentException(
-              "Passing RayObject to a direct call actor is not supported.");
-        }
         id = ((RayObject) arg).getId();
       } else {
         value = ObjectSerializer.serialize(arg);
-        if (!isDirectCall && value.data.length > LARGEST_SIZE_PASS_BY_VALUE) {
-          RayRuntime runtime = Ray.internal();
-          if (runtime instanceof RayMultiWorkerNativeRuntime) {
-            runtime = ((RayMultiWorkerNativeRuntime) runtime).getCurrentRuntime();
-          }
-          id = ((AbstractRayRuntime) runtime).getObjectStore()
-              .putRaw(value);
-          value = null;
+        if (value.data.length > LARGEST_SIZE_PASS_BY_VALUE) {
+          // Do nothing since we are not support pass by reference in direct call.
```

Comment:
I think this is already supported in Python?
does this mean that we pass everything by value? I think there might be problems with this.

Reply:
Yes, we pass all by value on the current master codebase.

If it has been supported in python, I'd like to post another PR to fix it, but this PR aims to remove direct call concept and do not change other logic any more.

Reply:
I created an issue to take a note. https://github.com/ray-project/ray/issues/7586

Reply:
ok, but this should be fixed ASAP.